libs/sdk-ui-charts/src/highcharts/chartTypes/warterfallChart/waterfallChartOptions.ts

```diff
@@ -0,0 +1,64 @@
+// (C) 2023 GoodData Corporation
+
+import { IColorAssignment } from "@gooddata/sdk-ui";
+import { IColorPalette, IMeasureGroupDescriptor, isRgbColor } from "@gooddata/sdk-model";
+import { getColorByGuid, getRgbStringFromRGB } from "@gooddata/sdk-ui-vis-commons";
+
+import { IChartConfig } from "../../../interfaces";
+import { ISeriesDataItem, ISeriesItem } from "../../typings/unsafe";
+
+export function buildWaterfallChartSeries(
+    series: ISeriesItem[],
+    chartConfig: IChartConfig,
+    colorAssignment: IColorAssignment,
+    colorPalette: IColorPalette,
+    emptyHeaderTitle: string,
+): ISeriesItem[] {
+    const isTotalSeriesEnabled = chartConfig.total?.enabled ?? false;
+    if (!isTotalSeriesEnabled) {
+        return series;
+    }
+
+    let seriesData: ISeriesDataItem[] = [];
```

Comment:
You can make this bit more readable if you would do opposite condition here:

````
if (!series || series.length === 0) {
   return [{ ...series[0], data: [] }];
}

// rest of the code here which no longer needs to be wrapped in `if` statement or using reasigning of variable which make the code less readable
````

Reply:
I updated this. Thank you!